### ISSUE TITLE
Changed behaviour for `url_parse`/`url_build` with paths

### ISSUE THREAD

**Issue by hongooi73** [r-lib/httr2#82]
httr2's URL parse/build treat paths differently to httr:

- If a URL doesn't contain a path, `httr2::url_parse` returns a NULL path, whereas `httr::parse_url` returns an empty string

```
r$> httr2::url_parse("http://resource")$path
NULL
r$> httr::parse_url("http://resource")$path
[1] ""
```

- A path is parsed with a starting `/` by `httr2::url_parse`, but not by `httr::parse_url`

```
r$> httr2::url_parse("https://resource/path")$path
[1] "/path"
r$> httr::parse_url("https://resource/path")$path
[1] "path"
```

- Passing a path without a starting `/` to `httr2::url_build` results in a malformed URL, which doesn't happen with `httr::build_url`

```
r$> url <- httr2::url_parse("http://resource")
    url$path <- "path"
    httr2::url_build(url)
[1] "http://resourcepath"
```

Possibly intended, but I thought I should mention it in case other people get tripped up.

**Comment by hongooi73** [comment]
Personally I find the older behaviour more intuitive, since `example.com` and `example.com/` should point to the same actual location. Similarly, there is only one reasonable way to interpret a path of "foo/bar", which is that it starts from the root.

**Comment by hadley** [comment]
I think the `NULL` vs `""` distinction is ok, as it does allow us to distinguish `https://x.com` from `https://x.com/`, which is probably not generally useful, seems like a nice distinction to have.

And I fixed the problem with missing `/` in #276. So I think overall this is ok now.
